Update the docs for V2 Changes

cloneofghosts opened the pull request:
Update the docs to be fully updated for V2 and finish HA documentation

- Fix https://github.com/Pirate-Weather/pirateweather/issues/214
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/146
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/211
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/231
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/213
- Fix https://github.com/Pirate-Weather/pirateweather/issues/215

alexander0042 commented:
Great, just pushed another set of additions to the V2.0 docs branch, which should round out all those updates. 

One thing I was thinking about was what sort of stuff should go in the HA docs that isn't already included? I added some stuff on alerts, but are there more FAQ's that should be added? Or something more fundamental?

cloneofghosts commented:
Was looking through the changes and I don't see anything specific in the HA docs about the alerts sensor. Maybe I missed it but it looks like it's not there? For the HA docs I'd include a link to the data points section in the API documentation. I would document any specific HA things in there that don't apply to the regular API. Be it as FAQs or just regular documentation.

The only other thing that is needed here is updating the OpenAPI specification as it looks like it was missed in these changes.

cloneofghosts commented on docs/API.md:
```diff
@@ -472,25 +460,25 @@ The time of the minimum "feels like" temperature during the daytime, from 6:00 a
 
 #### humidity
-Relative humidity expressed as a value between 0 and 1 inclusive. This is a percentage of the actual water vapor in the air compared to the total amount of water vapor that can exist at the current temperature. [See this resource for more information.](https://www.sciencedirect.com/topics/agricultural-and-biological-sciences/relative-humidity)
+Relative humidity expressed as a value between 0 and 1 inclusive. This is a percentage of the actual water vapor in the air compared to the total amount of water vapour that can exist at the current temperature. [See this resource for more information.](https://www.sciencedirect.com/topics/agricultural-and-biological-sciences/relative-humidity)
```
There is another instance of vapor that needs changed here

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -0,0 +1,54 @@
+Several models are used to produce the forecast. They are all hosted on [AWS's Open Data Platform](https://registry.opendata.aws/collab/noaa/), and the fantastic [Herbie package](https://github.com/blaylockbk/Herbie) is used to download and perform initial processing for all of them.    
+
+#### NBM
+The National Blend of Models [(NBM)](https://vlab.noaa.gov/web/mdl/nbm) is a calibrated blend of both NOAA and non-NOAA weather models from around the world. Running every hour for about days, the NBM produces a forecast that aims to leverage strengths from each of the source models, as well as providing some probabilistic forecasts. For most weather elements in the US and Canada, this is the pri ...
```
Missing the number of days

cloneofghosts commented:
Also as per [this comment](https://github.com/Pirate-Weather/pirate-weather-ha/issues/230#issuecomment-2097050507) maybe you should clarify where the storm distance is measured from?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
 
-|Parameter |Currently   |Minutely   |Hourly   |Daily   |
+|Parameter 	|Currently   |Minutely   |Hourly/ Daily   |
 |---|---|---|---|---|
```
Remove extra row to fix table formatting

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
```
Shouldn't this be HRRR > NBM > GEFS for all sections?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|fireIndex    			|NBM   			  		|N/A   	|NBM   			 			|
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Shouldn't the currently section be N/A? There is no `iceAccumulaton` field on the currently section.

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|fireIndex    			|NBM   			  		|N/A   	|NBM   			 			|
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Shouldn't the currently section be N/A? There is no `liquidAccumulaton` field on the currently section.

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
```
Shouldn't it be HRRR > GFS? Is NBM data used at all?

alexander0042 replied:
Storm bearing is always 100% GFS for now- it's a pretty heavy computational lift to calculate these two, and so it's much easier to just do it for one model every 6-hours

cloneofghosts replied:
Okay good to know. I remember seeing a in the nearest storm issue that you were going to use HRRR but it must have changed at some point.

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
```
Shouldn't it be HRRR > GFS? Is NBM data used at all?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
+|precipIntensityError	|GEFS					|GEFS	|GEFS						|	
+|precipProbability  	|NBM > GEFS 			|NBM 	|NBM > GEFS					|
+|precipType   			|NBM > HRRR > GEFS 		|HRRR  	|NBM > HRRR > GEFS			|
```
Shouldn't the currently section be HRRR > NBM > GEFS  and the minutely section HRRR > NBM > GEFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
+|precipIntensityError	|GEFS					|GEFS	|GEFS						|	
+|precipProbability  	|NBM > GEFS 			|NBM 	|NBM > GEFS					|
```
Shouldn't this always be NBM > GEFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|precipProbability  	|NBM > GEFS 			|NBM 	|NBM > GEFS					|
+|precipType   			|NBM > HRRR > GEFS 		|HRRR  	|NBM > HRRR > GEFS			|
+|pressure   			|HRRR > GFS   			|N/A	|HRRR > GFS 				|
+|snowAccumulation   	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS 	|
```
Shouldn't the currently section be N/A? There is no `snowAccumulaton` field on the currently section.

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|precipType   			|NBM > HRRR > GEFS 		|HRRR  	|NBM > HRRR > GEFS			|
+|pressure   			|HRRR > GFS   			|N/A	|HRRR > GFS 				|
+|snowAccumulation   	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS 	|
+|smoke   				|NBM   					|N/A   	|NBM   						|
```
Should be HRRR and not NBM

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|pressure   			|HRRR > GFS   			|N/A	|HRRR > GFS 				|
+|snowAccumulation   	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS 	|
+|smoke   				|NBM   					|N/A   	|NBM   						|
+|temperature   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Shouldn't the currently section be HRRR > NBM > GFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|temperature   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|uvIndex   				|GFS   					|N/A   	|GFS   						|
+|visibility   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windBearing  			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Shouldn't the currently section be HRRR > NBM > GFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|uvIndex   				|GFS   					|N/A   	|GFS   						|
+|visibility   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windBearing  			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windGust   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Shouldn't the currently section be HRRR > NBM > GFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|visibility   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windBearing  			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windGust   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windSpeed   			|NBM > HRRR > GFS   	|N/A	|NBM > HRRR > GFS   		|
```
Shouldn't the currently section be HRRR > NBM > GFS?

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Shouldn't the currently section be N/A? There is no `precipAccumulaton` field on the currently section.

cloneofghosts commented on docs/DataSources.md:
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
-
+|apparentTemperature	|NBM > GFS				|N/A   	|NBM > GFS   			 	|
+|cloudCover   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|		
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Shouldn't the currently section be HRRR > NBM > GFS?

cloneofghosts reviewed (commented):
Just had another look through the data sources document and just wanted to confirm where the data is coming from. Thought that the comment would just show the line I added the comment to. If you click view reviewed changes it will show the comment better,

alexander0042 commented:
Incredible notes, these are great! I'll go through them and double check each one.

Also, note for myself- smoke is ins kg/m3 (the model default), but ug/m3 makes way more sense. Need the docs to report ug/m3 and multiply the response by 10^9

cloneofghosts commented:
I fixed the missed spelling change and I updated a few items in the table that I'm sure of but you should double check and make sure.

alexander0042 commented:
Like I said earlier, amazing edits!!! I went through and put them all in, as well as updating the data pipeline page with the new times. I'm going to supersede that page with the details on the new V2 processing flow, but since that's just for general interest I want to get this out first.

Also added some more details on the storm distance, and updated the changelog/ roadmap. 

Also really good comment on the OpenAPI specs. Apiable really likes having those, and it's good practice, so I'll add those in before we push this.

cloneofghosts commented:
@alexander0042 I added in the new sensors to the HA page and did some minor updates to the OpenAPI spec (mainly change data types on some items which were incorrect). Also the DataSources.md page will need to be added to the menu.

cloneofghosts commented:
@alexander0042 I've updated the OpenAPI spec if you want to double check and make sure everything is good there. I also added in the new sensors into the HA docs page. Is there anything from my last comment that you want to add or is it good to merge?

cloneofghosts commented:
I added the new sensors which I added yesterday and the template from the templating issue in HA. @alexander0042 I think things are good now if you could review so we can get this merged in.

alexander0042 commented:
Ok,  with that fire out I finally had a chance to review this! Read through it all and it's looking good, so I'm going to push to main and see how everything formats

alexander0042 reviewed (commented):
Great edits, looks good to me!

cloneofghosts commented:
Was just about to go and fix the changelog version but you beat me to it and added 2.0.6 and 2.0.7. The OpenAPI spec will need to have the versions changed but that's pretty minor.
